app/models/energy/programtype.py

```diff
@@ -15,11 +16,21 @@ class PeopleAbridged(BaseModel):
 
     name: str = Schema(
         ...,
-        regex=r'^[\s.A-Za-z0-9_-]*$',
         min_length=1,
         max_length=100
     )
 
+    @validator('name')
+    def check_name(cls, v):
+        try:
+            val = ''.join(i for i in v if ord(i) < 128)
+            val = re.sub(r'[,;!\n\t]', '', v)
+        except  TypeError:
+            raise TypeError('Invalid String')
+        val = val.strip()
+        assert len(v) > 0, 'Name contains no valid characters.'
+        assert len(v) <=100, 'Number of characters must be less than 100.'
+
     people_per_area: float = Schema(
```

Comment:
@chriswmackey Does the name field validation look okay?

Reply:
Thanks for doing this @tanushree04 . However, it looks like your code here is trying to reformat the name into a valid one rather than just checking whether the name is valid to begin with. I should have clarified that the original Python code I sent it intended to format a "display name" input by a user into a valid name that can be sent to EnergyPlus.  So I think your code for the validator here should look something more like this:
```
assert all(ord(i) < 128 for i in v), 'Name contains non ASCII characters.'
assert all(char not in v for char in (',', ';', '!', '\n', '\t')), 'Name contains an invalid character for EnergyPlus ( ,  ;  !  \n  \t).'
assert len(v) > 0, 'Name contains no characters.'
assert len(v) <=100, 'Number of characters in name must be less than 100.'
```